tables/core/array.py

```diff
@@ -136,8 +140,10 @@ def copy(self, newparent=None, newname=None,
             newparent = self.root._get_or_create_path(newparent, createparents)
         if self.__class__.__name__ == 'Array':
             create_function = newparent.create_array
-        else:
+        elif self.__class__.__name__ == 'CArray':
```

Comment:
Probably it would be saner to use isinstance to test the condition, but you need to reverse the order of tests because CArraiy *is* an Array and EArray *is* a CArray.